internal/apis/certmanager/validation/issuer_test.go

```diff
@@ -1059,13 +1060,23 @@ func TestValidateACMEIssuerDNS01Config(t *testing.T) {
 				field.Invalid(fldPath.Child("rfc2136", "nameserver"), ":53", "nameserver must be set in the form host:port where host is an IPv4 address, an enclosed IPv6 address or a hostname and port is an optional port number."),
 			},
 		},
-		"rfc2136 provider using case-camel in algorithm": {
+		"rfc2136 provider using deprecated HMACMD5 algorithm": {
 			cfg: &cmacme.ACMEChallengeSolverDNS01{
 				RFC2136: &cmacme.ACMEIssuerDNS01ProviderRFC2136{
 					Nameserver:    "127.0.0.1",
 					TSIGAlgorithm: "HmAcMd5",
 				},
 			},
+			errs:     []*field.Error{},
+			warnings: []string{deprecatedHMacMD5Value},
+		},
+		"rfc2136 provider using camelcased value for HMACSHA256 algorithm": {
+			cfg: &cmacme.ACMEChallengeSolverDNS01{
+				RFC2136: &cmacme.ACMEIssuerDNS01ProviderRFC2136{
+					Nameserver:    "127.0.0.1",
+					TSIGAlgorithm: "HmAcSHA256",
```

Comment:
I love that this test looks [sArCaStIc](https://knowyourmeme.com/memes/mocking-spongebob) :joy:

Reply:
ahhhh